File: c++/2024/Circle.cpp
#include <iostream>

using namespace std ;

class Circle {
	private:
		double radius ;

	public:
		Circle() {
			radius = 0.0 ;
		}

		Circle(double r) {
			radius = r ;
		}

		void setRadius(double r) { // setter
			radius = r ;
		}

		double getRadius() {
			return radius ;
		}

		double getArea() {
			return radius * radius * 3.14159 ;
		}

		double getDiameter() {
			return 2 * radius ;
		}

		double getCircumference() {
			return 2 * radius * 3.14159 ;
		}

} ;

int main(void) {
	double rad , rad2 ;

	cout << "Enter a radius: " ;
	cin >> rad ;

	Circle sample ;
	sample.setRadius(rad) ; // using no-arg constructor

	cout << "Area = " << sample.getArea() << "\n" ;
	cout << "Diameter = " << sample.getArea() << "\n" ;
	cout << "Circumference = " << sample.getCircumference() << "\n" ;

	cout << "Enter another radius: " ;
	cin >> rad2 ;

	Circle sample2(rad2) ; // parameterized constructor

	cout << "Area = " << sample2.getArea() << "\n" ;
	cout << "Diameter = " << sample2.getArea() << "\n" ;
	cout << "Circumference = " << sample2.getCircumference() << "\n" ;

	return 0 ;
}

File: rust/2025/circle.rs
use std::io;

struct Circle {
    radius: f32
}

impl Circle {
    fn get_diameter(&self) -> f32 {
        return &self.radius * 2.0 ;
    }

    fn get_area(&self) -> f32 {
        return &self.radius * &self.radius * 3.14159 ;
    }

    fn get_circumference(&self) -> f32 {
        return &self.radius * 2.0 * 3.14159 ;
    }
}

fn main() {
    println!("--- Circle Driver ---") ;
    let juno = Circle {radius : 4.28} ;
    println!("** Given Parameters **") ;
    println!("Radius = {}", juno.radius) ;
    println!("** Functions **") ;
    println!("Diameter = {}", juno.get_diameter()) ;
    println!("Area = {}", juno.get_area()) ;
    println!("Circumference = {}", juno.get_circumference()) ;
}
